plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1498,9 +1499,10 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
             throw new TrinoException(NOT_SUPPORTED, "This connector does not support adding not null columns");
         }
         Table icebergTable = catalog.loadTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName());
+        AtomicInteger nextFieldId = new AtomicInteger(1);
```

Comment:
I think we want to initialize this to `icebergTable.schema().highestFieldId() + 1`

Reply:
Because `icebergTable.updateSchema().addColumn()` will reassigned nested type's fields id, so I just initialize id to 1. But initialization as you say can indicate the meaning. I will modify it.